src/components/ble/NimbleController.cpp

```diff
@@ -141,10 +154,11 @@ void NimbleController::StartAdvertising() {
   fields.uuids128_is_complete = 1;
   fields.tx_pwr_lvl = BLE_HS_ADV_TX_PWR_LVL_AUTO;
 
-  rsp_fields.name = (uint8_t*) deviceName;
+  rsp_fields.name = reinterpret_cast<const uint8_t*>(deviceName);
```

Comment:
Why is it `reinterpret_cast`?

Reply:
you request I lose the C-style cast ni a previous review :-)

Reply:
Yes, I guess I should've been more clear, does `static_cast` not work?

Reply:
I have vague memories of trying that first and it did not work. I'll try it (again?) and report back.

Reply:
I did try a static_cast first and here is what I found.

Using a static_cast won't allow casting from a signed to unsigned type so I need to reinterpret. For some inexplicable reason the device name is a char* everywhere in nimble except the response fields where it is a const uint8_t*